src/Mutator/Removal/ArrayItemRemoval.php

```diff
@@ -116,7 +115,11 @@ public static function getDefinition(): ?Definition
      */
     public function mutate(Node $node): iterable
     {
-        Assert::allNotNull($node->items);
+        $parent = $node->getAttribute('parent');
+        // Don't mutate destructured values in foreach loops
+        if ($parent instanceof Node\Stmt\Foreach_ && $parent->valueVar === $node) {
```

Comment:
Shouldn't it be in the `canMutate()` method below? Since there we are deciding whether to mutate `Node` or not

Reply:
Sorry, I totally overlooked that method. An update is on its way.